lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2137,18 +2198,14 @@ instance Buildable e => LiftHandler (ErrSelectCoinsExternal e) where
             handler e
         ErrSelectCoinsExternalUnableToMakeSelection e ->
             handler e
-        ErrSelectCoinsExternalUnableToAssignInputs wid ->
-            apiError err500 UnexpectedError $ mconcat
-                [ "I was unable to assign inputs while generating a coin "
-                , "selection for the specified wallet: "
-                , toText wid
-                ]
-        ErrSelectCoinsExternalUnableToAssignOutputs wid ->
-            apiError err500 UnexpectedError $ mconcat
-                [ "I was unable to assign outputs while generating a coin "
-                , "selection for the specified wallet: "
-                , toText wid
-                ]
+        ErrSelectCoinsExternalUnableToAssignInputs e ->
+            apiError err403 UnableToAssignInputOutput $ mconcat
+                [ "Unable to assign inputs from coin selection: "
+                , pretty e]
+        ErrSelectCoinsExternalUnableToAssignOutputs e ->
+            apiError err403 UnableToAssignInputOutput $ mconcat
+                [ "Unable to assign outputs from coin selection: "
+                , pretty e]
```

Comment:
Two remarks here:

a) This is changing the behavior from a server error to a client error. We aren't necessarily against API breaking changes in principle but they should be backed with a rationale. I don't think this is right here, there's nothing that clients can do about this, getting in this situation is a programmer error meaning that we screwed up.

b) May we please keep the first-person tone to remain consistent with the rest of the server errors. It is much more friendly to get errors reported in the first-person :)